test/e2e/performanceprofile/functests/5_latency_testing/latency_testing.go

```diff
@@ -219,6 +225,33 @@ func clearEnv() {
 	os.Unsetenv(latencyTestCpus)
 }
 
+// checkHyperthreading determines if the BZ 2094046 test should run based on HT status.
+// Returns true if the test should run (HT enabled), false if it should be skipped (HT disabled).
+func checkHyperthreading() (bool, error) {
+	profile, err := profiles.GetByNodeLabels(testutils.NodeSelectorLabels)
+	if err != nil {
+		return false, fmt.Errorf("get performance profile: %w", err)
+	}
+
+	workerNodes, err := nodes.GetByLabels(testutils.NodeSelectorLabels)
+	if err != nil {
+		return false, fmt.Errorf("get worker nodes: %w", err)
+	}
+	workerNode := &workerNodes[0]
```

Comment:
why do we need to pick a random node which matches the labels? can't we just pick the node by name?